### PR TITLE
Implement decorator

### PR DESCRIPTION
**Pull request by Baroka-wp**
In this PR we implement the following : 

- [ ] Create a class Nameable.
- Implement a method called correct_name that will raise a NotImplementedError.

- [ ] Turn your Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

- [ ] Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the `correct_name` method of the `@nameable`.

- [ ] Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
- [ ] Create a class that inherits from the base Decorator class.
- Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
- [ ] Create a class that inherits from the base Decorator class.
- Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (DammyShittu) [inline capitalize_decorator.rb]: - [ ] We are required to capitalize the correct_name, not make it an uppercase. Kindly replace `.upcase` with `.capitalize`.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Baroka-wp,

## Highlights

✔️ Trimming of name
✔️ Base decorator

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: capitalize_decorator.rb (hunk 1 above)
Comment (Baroka-wp) [reply capitalize_decorator.rb]: ## Thank you @DammyShittu

**Review by DammyShittu — APPROVED** [review]
Hi @Baroka-wp,

## APPROVED 🟢

## Highlights

✔️ Descriptive PR
✔️ Correct name trimming and capitalization
✔️ Linters are passing

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Baroka-wp** [comment]
## THANK YOU @DammyShittu

**Pull request merged**